src/pkg/art/selectors/doublestar/selector.go

```diff
@@ -85,6 +96,32 @@ func (s *selector) Select(artifacts []*art.Candidate) (selected []*art.Candidate
 	return selected, nil
 }
 
+func (s *selector) tagSelectMatch(artifact *art.Candidate) (selected bool, err error) {
+	for _, t := range artifact.Tags {
+		matched, err := match(s.pattern, t)
+		if err != nil {
+			return false, err
+		}
+		if matched {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (s *selector) tagSelectExclude(artifact *art.Candidate) (selected bool, err error) {
```

Comment:
This logic seems has problem, the only case to return `false` is all the tags are excluded, right?